client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/MigrationDocumentationIT.java

```diff
@@ -155,6 +151,14 @@ public void onFailure(Exception e) {
                 }
             });
             //end::migration-request-async-execution
+            assertBusy(() -> assertFalse(client.exists(new GetRequest("index", "doc", "id"))));
```

Comment:
was this another problem? I wonder why this assertBusy is needed.

Reply:
Yes, previously the last code snippet was an async deletion and the test sometimes fails (or warn a thread leak) because of ongoing operation when shutting down the cluster. 

I swapped the async and syn deletion operation so that the async is executed first, and this assertBusy is here to wait for the async operation to finish. A CountDownLatch would have been better but since this is a documentation code snippet I didn't want the latch to appear in doc so I did it this way.

Reply:
I understand, that makes sense. Maybe add a tiny comment on why the assertBusy is there (the wait could be elsewhere but that would also appear in the docs snippet which is not good)